### PR TITLE
Fix merging hover results

### PR DESCRIPTION
**Pull request by yyoncho**
- this is causing issues in lsp-java when combined with spring-boot

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -1197,17 +1197,15 @@ the lists according to METHOD."
...
-                                                   (lsp:hover-contents result))))
-                                        (lsp-make-hover)
-                                        hovers))))
+                               (hovers (lsp-make-hover
+                                        :contents
+                                        (apply #'append (-map
+                                                         (-lambda ((&Hover :contents))
+                                                           (if (and (sequencep contents)
+                                                                    (not (stringp contents)))
+                                                               (append contents ())
```
Comment (nbfalcon) [inline lsp-mode.el:1206]: Why would you need to clone `contents` here? `append` is non-destructive.

File: lsp-mode.el [hunk 2]
```diff
@@ -1197,17 +1197,15 @@ the lists according to METHOD."
...
+                                                             (list contents)))
+                                                         hovers))))))
```
Comment (nbfalcon) [inline lsp-mode.el:1208]: Wouldn't hovers always be nil here, causing hovers to always be empty in this code path? This also seems present in the old code, but I don't see it.

File: lsp-mode.el [hunk 3]
```diff
@@ -1197,17 +1197,15 @@ the lists according to METHOD."
...
-                                                   (if (and (sequencep contents)
-                                                            (not (stringp contents)))
-                                                       contents
-                                                     (list contents))
-                                                   (lsp:hover-contents result))))
-                                        (lsp-make-hover)
-                                        hovers))))
+                               (hovers (lsp-make-hover
+                                        :contents
+                                        (apply #'append (-map
```
Comment (nbfalcon) [inline lsp-mode.el:1202]: `-mapcat`

**Review by nbfalcon — COMMENTED** [review]
Aside from a minor nitpick (`-mapcat`) and some confusion on why this works on my side, LGTM.

**Comment by yyoncho** [comment]
Fixed comments.

**Review by nbfalcon — APPROVED** [review]

File: lsp-mode.el (hunk 2 above)
Comment (nbfalcon) [reply lsp-mode.el]: Never mind, I forgot how `pcase` works for a second...

**Review by ericdallo — APPROVED** [review]

**Review by jcs090218 — APPROVED** [review]

**Pull request merged**